addons/pos_self_order/static/src/app/kiosk_style.js

```diff
@@ -1,52 +1,83 @@
-export function insertKioskStyle(primaryBgColor, primaryTextColor) {
+export function insertKioskStyle(primaryBgColor) {
     const style = document.createElement("style");
-    style.textContent = generateKioskCSS(primaryBgColor, primaryTextColor);
+    style.textContent = generateKioskCSS(primaryBgColor);
     document.head.appendChild(style);
 }
 
-function generateKioskCSS(primaryBg, primaryText = "#fff") {
-    if (!primaryBg || primaryBg === "#875A7B") {
-        primaryBg = "#714B67";
+function generateKioskCSS(companyPrimaryColor) {
+    let bgPrimary = companyPrimaryColor;
+    if (!bgPrimary || bgPrimary === "#875A7B") {
+        bgPrimary = "#714B67";
     }
-    const activeBG = shadeColor(primaryBg, 0.2);
-    const primaryRGB = hexToRgb(primaryBg);
-    const hoverFocusRGB = hexToRgb(mixColors(primaryText, primaryBg, 0.15));
-    const kioskBG = mixColors("#fff", primaryBg, 0.85);
+    const luminance = getLuminance(bgPrimary);
+    const isLightBackground = luminance > 0.55;
+    const shadedPrimary = shadeColor(bgPrimary, 0.5);
+
+    const textBgPrimary = isLightBackground
+        ? shadeColor(bgPrimary, 0.7)
+        : mixColors("#FFFFFF", bgPrimary, 0.85);
```

Comment:
This is subtle but I would increase the mixColor to .95 or even use plain white, this is what bootstrap does with the btn-color (#FFF / #000)

Here with .85, it feels a bit "disabled" 
![image](https://github.com/user-attachments/assets/f5d2ecca-bb62-452d-8521-9380f31d482d)

Here with .95 
![image](https://github.com/user-attachments/assets/0ffb4a2c-6056-40ba-92d4-d8f7e6114470)